src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,1199 @@
+        out = F.relu_(out)
+        return out
+
+
+class TVPVisionBottleneckBlock(TVPVisionBlockBase):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        bottleneck_channels,
+        stride=1,
+        num_groups=1,
+        dilation=1,
+    ):
+        super().__init__(in_channels, out_channels, stride)
+
+        if in_channels != out_channels:
+            self.shortcut = TVPVisionConv2d(
+                in_channels,
+                out_channels,
+                kernel_size=1,
+                stride=stride,
+                bias=False,
+                norm=TVPVisionFrozenBatchNorm2d(out_channels),
+            )
+        else:
+            self.shortcut = None
+
+        self.conv1 = TVPVisionConv2d(
+            in_channels,
+            bottleneck_channels,
+            kernel_size=1,
+            stride=stride,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(bottleneck_channels),
+        )
+
+        self.conv2 = TVPVisionConv2d(
+            bottleneck_channels,
+            bottleneck_channels,
+            kernel_size=3,
+            stride=1,
+            padding=dilation,
+            bias=False,
+            groups=num_groups,
+            dilation=dilation,
+            norm=TVPVisionFrozenBatchNorm2d(bottleneck_channels),
+        )
+
+        self.conv3 = TVPVisionConv2d(
+            bottleneck_channels,
+            out_channels,
+            kernel_size=1,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(out_channels),
+        )
+
+        for layer in [self.conv1, self.conv2, self.conv3, self.shortcut]:
+            if layer is not None:  # shortcut can be None
+                c2_msra_fill(layer)
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = F.relu_(out)
+
+        out = self.conv2(out)
+        out = F.relu_(out)
+
+        out = self.conv3(out)
+
+        if self.shortcut is not None:
+            shortcut = self.shortcut(x)
+        else:
+            shortcut = x
+
+        out += shortcut
+        out = F.relu_(out)
+        return out
+
+
+class TVPVisionBackbone(nn.Module):
+    def __init__(self, config):
+        super(TVPVisionBackbone, self).__init__()
+        self.config = config
+        self.stem = TVPVisionBasicStem(
+            in_channels=config.resnets_stem_input_channels,
+            out_channels=config.resnets_stem_out_channels,
+        )
+        if config.backbone_freeze_at >= 1:
+            for p in self.stem.parameters():
+                p.requires_grad = False
+
+        current_stride = self.stem.stride
+        self._out_feature_strides = {"stem": current_stride}
+        self._out_feature_channels = {"stem": self.stem.out_channels}
+
+        stages = self.build_backbone(config)
+
+        self.stages_and_names = []
+        for i, blocks in enumerate(stages):
+            for block in blocks:
+                assert isinstance(block, TVPVisionBlockBase), block
```

Comment:
We don't have asserts in our modeling code - this should be removed or made an exception